src/Files.App/Views/HomePage.xaml.cs

```diff
@@ -121,6 +121,7 @@ public void ReloadWidgets()
 			{
 				Widgets.ViewModel.InsertWidget(new(recentFilesWidget, (value) => UserSettingsService.PreferencesSettingsService.RecentFilesWidgetExpanded = value, () => UserSettingsService.PreferencesSettingsService.RecentFilesWidgetExpanded), 4);
 
+				recentFilesWidget.AppInstance = AppInstance;
```

Comment:
This looks like the cause of #12160 🤔

Reply:
I tested just after this PR was merged and the CPU usage is not high.
![image](https://user-images.githubusercontent.com/66369541/233820227-cadd00f3-e3a7-4ad5-858a-a8e16c7db42e.png)